src/patternfly/components/Menu/examples/Menu.md

```diff
@@ -0,0 +1,812 @@
+          {{/menu-item-text}}
+        {{/menu-item-main}}
+        {{#> menu-item-description}}
+          Description
+        {{/menu-item-description}}
+      {{/menu-item}}
+    {{/menu-list-item}}
+    {{#> menu-list-item}}
+      {{#> menu-item}}
+        {{#> menu-item-main}}
+          {{#> menu-item-icon}}
+            <i class="fas fa-code-branch" aria-hidden="true"></i>
+          {{/menu-item-icon}}
+          {{#> menu-item-text}}
+            Action 3
+          {{/menu-item-text}}
+        {{/menu-item-main}}
+        {{#> menu-item-description}}
+          Nunc non ornare ex, et pretium dui. Duis nec augue at urna elementum blandit tincidunt eget metus. Aenean sed metus id urna dignissim interdum. Aenean vel nisl vitae arcu vehicula pulvinar eget nec turpis. Cras sit amet est est. 
+        {{/menu-item-description}}
+      {{/menu-item}}
+    {{/menu-list-item}}
+  {{/menu-body}}
+{{/menu}}
+```
+
+### With-actions
+```hbs
+{{#> menu}}
+  {{#> menu-body}}
+    {{#> menu-group}}
+      {{#> menu-group-title}}
+        Actions
+      {{/menu-group-title}}
+      <ul>
+        {{#> menu-list-item}}
+          {{#> menu-item}}
+            {{#> menu-item-main}}
+              {{#> menu-item-text}}
+                Item 1 
+              {{/menu-item-text}}
+              {{> menu-item-external-icon}}
+            {{/menu-item-main}}
+            {{#> menu-item-description}}
+              This is a description
+            {{/menu-item-description}}
+          {{/menu-item}}
+          {{#> menu-item-action}}
+            <i class="fas fa-ellipsis-v" aria-hidden="true"></i>
+          {{/menu-item-action}}
+        {{/menu-list-item}}
+        {{#> menu-list-item}}
+          {{#> menu-item}}
+            {{#> menu-item-main}}
+              {{#> menu-item-text}}
+                Item 2
+              {{/menu-item-text}}
+              {{> menu-item-external-icon}}
+            {{/menu-item-main}}
+          {{/menu-item}}
+          {{#> menu-item-action}}
+            <i class="fas fa-bell" aria-hidden="true"></i>
+          {{/menu-item-action}}
+        {{/menu-list-item}}
+        {{#> menu-list-item}}
+          {{#> menu-item}}
+            {{#> menu-item-main}}
+              {{#> menu-item-text}}
+                Item 3
+              {{/menu-item-text}}
+              {{> menu-item-external-icon}}
+            {{/menu-item-main}}
+            {{#> menu-item-description}}
+              This is a description
+            {{/menu-item-description}}
+          {{/menu-item}}
+          {{#> menu-item-action}}
+            <i class="fas fa-clipboard" aria-hidden="true"></i>
+          {{/menu-item-action}}
+        {{/menu-list-item}}
+        {{#> menu-list-item}}
+          {{#> menu-item}}
+            {{#> menu-item-main}}
+              {{#> menu-item-text}}
+                Item 4
+              {{/menu-item-text}}
+              {{> menu-item-external-icon}}
+              {{> menu-item-select-icon}}
```

Comment:
This needs to behave similarly to the external link icon so that its appearance doesn't shift the menu around when it shows/hides. I think we would hide it visually like the external link icon, then show it when the item is `pf-m-selected`

![Aug-18-2020 10-41-07](https://user-images.githubusercontent.com/35148959/90534918-efc3d780-e13f-11ea-9c1c-5d4f3e8f8852.gif)